AAT_gae/src/main/java/com/aat/restlet/AttendanceResource.java

```diff
@@ -186,10 +189,14 @@ private AttendanceRecord retrieveAttendanceRecord(Long userId, Long groupId){
 		
 		List<Ref<AttendanceRecord>> refAttendances =  student.getGroups();
 		for (Ref<AttendanceRecord> refAttendance : refAttendances){	
-			AttendanceRecord ar = refAttendance.get();
-			if (ar != null) {
-				long parentId = ar.getParent().getId();
-				if (groupId == parentId){
+			AttendanceRecord ar = refAttendance.getValue();
+			if (ar!=null){
+				Key<Group> keygroup = ar.getParent();
+				Group group = ObjectifyService.ofy().load().key(keygroup).now();
+				long storedGroupId = keygroup.getId();
+				long storedCourseId = group.getParent().getId();
```

Comment:
Can't we do this without loading the group from objectify?

Just call ar.getParent().getId() to get the groupId and ar.getParent().getParent().getId() to get the courseId?

See http://static.javadoc.io/com.googlecode.objectify/objectify/5.1.14/com/googlecode/objectify/Key.html#getParent--

This will decrease the number of queries we have to make.